src/scripts/main.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, function handler(_event) {
+      element.removeEventListener(eventName, handler);
+
+      const message = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`;
```

Comment:
The message construction correctly uses template literals to include the event name and element details. Ensure that `element.id` is defined, as it might be `undefined` if the element does not have an ID.